packages/dd-trace/test/encode/0.4.spec.js

```diff
@@ -38,6 +38,7 @@ describe('encode', () => {
       start: 123,
       duration: 456
     }]
+    sinon.spy(encoder._traceBytes, '_resize')
```

Comment:
Why is a spy on an internal method needed? Can't we just validate that it's not flushing garbage data?

Reply:
if we ever change the chunk size, this test can inadvertently start testing nothing, but not failing.

Reply:
Any way to do this without spying on the internal API? It will soon be impossible to write tests this way, so we will have to stop relying on this in any case.

Reply:
well you could check that the reference of `encoder._traceBytes` has changed. It'd still be checking the `_traceBytes` internal, but what this is test is protecting us against is precisely using that internal wrongly